### DISCUSSION TITLE
Applying HiFi-GAN to long sequences

### DISCUSSION THREAD

**Discussion by bryanhpchiang** [NVIDIA-NeMo/NeMo#7169] [Q&A]
Here's a basic question: so HiFi-GAN is trained with a segment size of 8192 by default (8192 waveform samples), which corresponds to around `8192 / hop_size ` Mel frames. 

What happens when we want to turn more than `8192 / hop_size` Mel frames into audio? Is the generator applied repeatedly? Thanks!

**Comment by XuesongYang** [comment]
I'm not entirely certain about your question, but you could opt to specify a longer audio segment or longer mel frames during the inference stage.

**Comment by Ziyi6** [comment]
@bryanhpchiang Did you find out something regarding the segment size?
